src/tests/Autosuggest.js

```diff
@@ -412,6 +426,106 @@ describe('Autosuggest', function() {
+
+      it('should call onSuggestionFocused when suggestion focused using Up/Down keys', function() {
+        clickDown();
+        expect(onSuggestionFocused).toBeCalledWith('Mill Park');
+      });
+
+      it('should not call onSuggestionFocused when first suggestion is focused and Up key is clicked', function() {
+        clickDown();
+        onSuggestionFocused.mockClear();
+        clickUp();
+        expect(onSuggestionFocused).not.toBeCalled();
+      });
+
+      it('should not call onSuggestionFocused when last suggestion is focused and Down key is clicked', function() {
+        clickDown();
+        clickDown();
+        onSuggestionFocused.mockClear();
+        clickDown();
+        expect(onSuggestionFocused).not.toBeCalled();
+      });
+    });
+  });
+
+  describe('onSuggestionUnfocused', function() {
+    beforeEach(function() {
+      onSuggestionFocused.mockClear();
+      onSuggestionUnfocused.mockClear();
+      createAutosuggest(
+        <Autosuggest suggestions={getSuburbObjects}
+                     suggestionRenderer={renderSuburbObject}
+                     suggestionValue={getSuburbObjectValue}
+                     onSuggestionFocused={onSuggestionFocused}
+                     onSuggestionUnfocused={onSuggestionUnfocused} />
+      );
+      setInputValue('m');
+    });
+
+    describe('Mouse interactions', function() {
+
+      it('should call onSuggestionUnfocused when suggestion unfocused using mouse', function() {
+        mouseOverFromInputToSuggestion(0);
+        mouseOverBetweenSuggestions(0, 1);
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
+      });
+
+      it('should call onSuggestionUnfocused when clicking outside and suggestion is focused', function() {
+        mouseOverFromInputToSuggestion(0);
+        clickOutside();
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
+      });
+
+      it('should not call onSuggestionUnfocused when clicking outside and no suggestion focused', function() {
+        clickOutside();
+        expect(onSuggestionUnfocused).not.toBeCalled();
+      });
+    });
+
+    describe('Keyboard interactions', function() {
+
+      it('should call onSuggestionUnfocused when suggestion unfocused using Up/Down keys', function() {
+        clickDown();
+        clickDown();
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
+      });
+
+      it('should call onSuggestionUnfocused when ESC key pressed and suggestion is focused', function() {
+        clickDown();
+        clickEscape();
+        expect(onSuggestionUnfocused).toBeCalledWith({ suburb: 'Mill Park', postcode: '3083' });
+      });
+
+      it('should not call onSuggestionUnfocused when ESC key pressed and no suggestion focused', function() {
+        clickEscape();
+        expect(onSuggestionUnfocused).not.toBeCalled();
+      });
+    });
+  });
```

Comment:
I wonder what do you expect to happen when suggestion is selected via mouse click or via clicking Enter? Shall we call `onSuggestionUnfocused` in both cases? Could you please add tests for those?